### DISCUSSION TITLE
How do I replace my private email with my no-reply github email on all old repos in all old commits?

### DISCUSSION THREAD

**Discussion by ddjerqq** [community/community#135009] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

Hi, when I originally signed up on github, I used my private email and I do not want to leak it on the commits publicly. I'd like to make it private if possible, but also retain all of my commits on all mine and outside repos.

I have found a way to replace the author on one repo using this command:
`git rebase -r --root --exec 'git commit --amend --no-edit --reset-author'` however, this is a really manual process, and it makes the commits act like they were made the day I execute this command, basically it purges the date. 

I know its petty, but I love seeing my contrib graph all green. 

Has anyone had a similar situation? could there be a script that will directly access a github PAT and access all repos and recommit all commits with the correct private github's noreply email?

I had known about this privacy issue for some time but always delayed it.

Any help is appreciated, thanks!

**Comment by ddjerqq** [comment]
here is what i have currently

```sh
#!/bin/sh

read -p "Do you want to amend the author of commits? (yes/no): " response
if [ "$response" != "yes" ]; then
    echo "Operation aborted."
    exit 1
fi

git rebase -r --root --exec 'git commit --amend --no-edit --reset-author --date="$(git log -1 --format=%ai)"'

read -p "Do you want to push the commits? (yes/no): " response
if [ "$response" != "yes" ]; then
    echo "Operation aborted."
    exit 1
fi

git push origin --force --all
```

**Comment by Pilot-61** [comment]


**Reply by ddjerqq** [reply]
this really looks like a reply i'd get our of chatgpt. which does not even take into account the shell script which I already have. which is not what i need. i have access to chatgpt too, this reply is worthless.
